gtfs/spec/en/reference.md

```diff
@@ -405,13 +408,34 @@ Trip planners normally calculate transfer points based on the relative proximity
 |  to_stop_id | **Required** | The **to_stop_id** field contains a stop ID that identifies a stop or station where a connection between routes ends. Stop IDs are referenced from the [stops.txt](#stopstxt) file. If the stop ID refers to a station that contains multiple stops, this transfer rule applies to all stops in that station. |
+|  from_route_id | **Optional** | The from_route_id field can contain a route_id (as specified by [routes.txt](#routestxt)), reducing the scope to which the given transfer applies. If from_route_id is specified, the transfer will only apply to the arriving trip with the given route id, at the given from_stop_id.  |
+|  to_route_id | **Optional** | The to_route_id field can contain a route_id (as specified by [routes.txt](#routestxt)), reducing the scope to which the given transfer applies. If to_route_id is specified, the transfer will only apply to the departing trip with given route id, at the given to_stop_id. |
+|  from_trip_id | **Optional** | The from_trip_id and to_trip_id fields can contain a trip_id, as specified by [trips.txt](#tripstxt). If from_trip_id is given, from_route_id is ignored. If from_trip_id is specified, the transfer will only apply to the arriving trip with the given trip id, at the given from_stop_id. |
+|  to_trip_id | **Optional** | The to_trip_id field can contain a trip_id, as specified by [trips.txt](#tripstxt). If to_trip_id is given, to_route_id is ignored. If to_trip_id is specified, the transfer will only apply to the departing trip with the given trip id, at the given to_stop_id.  |
 |  transfer_type | **Required** | The **transfer_type** field specifies the type of connection for the specified (from_stop_id, to_stop_id) pair. Valid values for this field are: |
 |   |  | * **0** or **(empty)** - This is a recommended transfer point between two routes. |
-|   |  | * **1** - This is a timed transfer point between two routes. The departing vehicle is expected to wait for the arriving one, with sufficient time for a passenger to transfer between routes. |
+|   |  | * **1** - This is a timed transfer point between two vehicles. The departing vehicle is expected to wait for the arriving one, with sufficient time for a passenger to transfer between routes. |
 |   |  | * **2** - This transfer requires a minimum amount of time between arrival and departure to ensure a connection. The time required to transfer is specified by **min_transfer_time**. |
-|   |  | * **3** - Transfers are not possible between routes at this location. |
+|   |  | * **3** - Transfers are not possible. |
+|   |  | * **4** - Passengers can transfer from one route to another by staying onboard the same vehicle (an "in-seat transfer") in cases where sequential trips are made by the same vehicle (same **block_id**).  In-seat transfer between these two stops / routes / trips is allowed. |
```

Comment:
Does the reference to block_id mean that block_id must be included to use this transfer_type? 

Which of the following are valid uses of transfer_type 4?
A. transfers.txt contains route_id's and stop_id's. The trips on the routes do have block_id's. 
B. transfers.txt contains trip_id's. The trips on the routes do have block_id's. 
C. transfers.txt contains trip_id's. The trips on the routes DO NOT have block_id's. 
D. transfers.txt contains route_id's and stop_id's. The trips on the routes DO NOT have block_id's.

Reply:
I agree with @aybrd that all these uses should be valid.

@abyrd: I think you're right that block_id should not be required in the case that transfer_type=4 is specified, but I still anticipate that feed producers may want to supply block_id to propagate vehicle delays as you suggest, to configure real-time passenger information systems (e.g. @TheTransitClock), and for other purposes.

Reply:
The text does need to be updated to remove the reference to block_id, which only applies to trips. This PR includes elements of the transfer specificity proposal, which would allow transfer_type 4 transfers to be created between routes or stops, not just trips. It definitely makes sense for a producer using transfer_type 4 to still include block_ids for realtime propagation purposes, but it doesn't seem relevant to the present section of the spec.